cmd/http-traffic-recorder.go

```diff
@@ -0,0 +1,84 @@
+/*
+ * MinIO Cloud Storage, (C) 2019 MinIO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strings"
+)
+
+// records the incoming bytes from the underlying request.Body.
+type recordTrafficRequest struct {
+	// wrapper for the underlying req.Body.
+	reader io.Reader
+	// incoming request.
+	request *http.Request
+}
+
+// Records the bytes read.
+func (r *recordTrafficRequest) Read(p []byte) (n int, err error) {
+	n, err = r.reader.Read(p)
+	globalConnStats.incInputBytes(n)
+	if strings.HasPrefix(r.request.URL.Path, "/minio") {
```

Comment:
Note that "/minio" is not S3 traffic i.e the logic here should be opposite of what it is here.

Reply:
Oh. I understood wrongly that all outbound/RPC calls are s3requests. Changed it. Thanks @krishnasrinivas. PTAL